src/main/java/ch/njol/skript/expressions/ExprItemsIn.java

```diff
@@ -40,109 +42,134 @@
-		"Please note that the positions of the items in the inventory are not saved, only their order is preserved."})
-@Examples({"loop all items in the player's inventory:",
-		"	loop-item is enchanted",
-		"	remove loop-item from the player",
-		"set {inventory::%uuid of player%::*} to items in the player's inventory"})
-@Since("2.0")
+@Description({
+	"All items or specific type(s) of items in an inventory. Useful for looping or storing in a list variable.",
+	"Please note that the positions of the items in the inventory are not saved, only their order is preserved."
+})
+@Examples({
+	"loop all items in the player's inventory:",
+	"\tloop-item is enchanted",
+	"\tremove loop-item from the player",
+	"set {inventory::%uuid of player%::*} to items in the player's inventory"
+})
+@Since("2.0, INSERT VERSION (specific types of items)")
 public class ExprItemsIn extends SimpleExpression<Slot> {
+
 	static {
-		Skript.registerExpression(ExprItemsIn.class, Slot.class, ExpressionType.PROPERTY, "[(all [[of] the]|the)] items ([with]in|of|contained in|out of) (|1¦inventor(y|ies)) %inventories%");
+		Skript.registerExpression(ExprItemsIn.class, Slot.class, ExpressionType.PROPERTY,
+			"[all [[of] the]|the] (items|%-itemtypes%) ([with]in|of|contained in|out of) [1:inventor(y|ies)] %inventories%"
+		);
 	}
-	
+
 	@SuppressWarnings("null")
-	private Expression<Inventory> invis;
-	
-	@SuppressWarnings({"unchecked", "null"})
+	private Expression<Inventory> inventories;
+
+	@Nullable
+	private Expression<ItemType> types;
+
 	@Override
+	@SuppressWarnings({"unchecked", "null"})
 	/*
 	 * the parse result will be null if it is used via the ExprInventory expression, however the expression will never
 	 * be a variable when used with that expression (it is always a anonymous SimpleExpression)
 	 */
-	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, @Nullable final ParseResult parseResult) {
-		invis = (Expression<Inventory>) exprs[0];
-		if (invis instanceof Variable && !invis.isSingle() && parseResult.mark != 1)
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		types = (Expression<ItemType>) exprs[0];
+		inventories = (Expression<Inventory>) exprs[1];
+		if (inventories instanceof Variable && !inventories.isSingle() && parseResult.mark != 1)
 			Skript.warning("'items in {variable::*}' does not actually represent the items stored in the variable. Use either '{variable::*}' (e.g. 'loop {variable::*}') if the variable contains items, or 'items in inventories {variable::*}' if the variable contains inventories.");
 		return true;
 	}
-	
-	@SuppressWarnings("null")
+
 	@Override
+	@SuppressWarnings("null")
 	protected Slot[] get(final Event e) {
-		final ArrayList<Slot> r = new ArrayList<>();
-		for (final Inventory invi : invis.getArray(e)) {
+		ArrayList<Slot> r = new ArrayList<>();
```

Comment:
```suggestion
		ArrayList<Slot> result = new ArrayList<>();
```
Should have a better name